### ISSUE TITLE
feature request: auto repair

### ISSUE THREAD

**Issue by ohitstom** [Vencord/Installer#103]
would it be possible to setup an nssm service and have vencord check if its installed and if not reinstall/repair on system boot? or something similar?

**Comment by JayBe13** [comment]
LOL I just came here to ask the same question.

**Comment by Vendicated** [comment]
this is not a bad idea, but i don't like the implementation

It would be much simpler to
- download the cli
- create a shortcut (or .desktop file on linux) that calls `vencord-installer-cli --install --location C:\Path\To\Discord && discord`. the user can then just use that shortcut instead of the normal discord shortcut and it'll always patch discord before launching it

**Comment by ohitstom** [comment]
> this is not a bad idea, but i don't like the implementation
> 
> It would be much simpler to
> - download the cli
> - create a shortcut (or .desktop file on linux) that calls `vencord-installer-cli --install --location C:\Path\To\Discord && discord`. the user can then just use that shortcut instead of the normal discord shortcut and it'll always patch discord before launching it

it could also very easily be made a service using that exact same set of steps, and then the user doesn't have to do it.

**Comment by lewisakura** [comment]
installing a system service is extremely undesirable and i will outright say the worst possible solution - it usually requires admin privileges and that should be completely unnecessary to install a client mod...

**Comment by ohitstom** [comment]
> installing a system service is extremely undesirable and i will outright say the worst possible solution - it usually requires admin privileges and that should be completely unnecessary to install a client mod... 

it does not require admin services to create a service, nor would this be undesirable. It could be a simple additional option for those  who don't want to have to redownload and repair vendors every discord update.

Many other client mods, self hosted, and various other projects rely on services.

**Comment by lewisakura** [comment]
> it does not require admin services to create a service

windows does, there's no concept of a set of user services and every service should run under a system service account (either one of the existing ones or something under the `NT SERVICE\` domain), not a user account. you can't even start and stop services without admin, let alone install one.

> nor would this be undesirable

you can't.. state that our own viewpoint is wrong. ***we*** see it as undesirable and we won't be doing it like this. we have our own plans for implementation that involves shortcuts like vee previously mentioned.

> Many other client mods, self hosted, and various other projects rely on services.

just because another project relies on services doesn't mean we should too. also, what on earth is using services to maintain a discord client mod? are you sure it's not just an app that runs on startup (which is VERY MUCH not a service)?

**Comment by ohitstom** [comment]
> > it does not require admin services to create a service
> 
> windows does, ~~there's no concept of a set of user services~~ (i am wrong, but it still requires admin to install and configure) and every service should really run under a service account (either one of the existing ones or something under the `NT SERVICE\` domain), not a user account. you can't even start and stop services without admin, let alone install one.
> 
> > nor would this be undesirable
> 
> you can't.. state that our own viewpoint is wrong. ***we*** see it as undesirable and we won't be doing it like this. we have our own plans for implementation that involves shortcuts like vee previously mentioned.
> 
> > Many other client mods, self hosted, and various other projects rely on services.
> 
> just because another project relies on services doesn't mean we should too. also, what on earth is using services to maintain a discord client mod? are you sure it's not just an app that runs on startup (which is VERY MUCH not a service)?

so what you expect users to overwrite every discord shortcut manually just to have vencord auto repair? if anythings undesirable it's exactly that

**Comment by lewisakura** [comment]
no, we create a shortcut for you as part of the install step, you dont have to do it yourself

**Comment by ohitstom** [comment]
> no, we create a shortcut for you as part of the install step, you dont have to do it yourself

what about users that have multiple shortcuts, shortcuts on the taskbar, start menu, file explorer? you expect them to manually change them all

**Comment by lewisakura** [comment]
we dont expect users to change *anything*, we're giving them a shortcut to use instead, they can just repin it to the taskbar or whatever.

we will NOT do a service because installing one varies on the operating system and varies in permission level, plus overall it is a very strange thing for a client mod installer to run a service on someone's system in the background. doing it on linux, for instance, will be a nightmare because different service managers exist and we'd have to write a user service unit for each one to ensure full compatibility. usually it's systemd, but sometimes it isn't. doing it on windows still requires admin and there is no way of getting around that.

i'm just going to lock this for now because this is getting awfully silly when nothing productive is happening in terms of progress on implementing the feature - discussion can continue in #vencord-dev on discord if desired.
